src/porepy/applications/md_grids/model_geometries.py

```diff
@@ -155,3 +155,53 @@ def meshing_arguments(self) -> dict:
             "cell_size_min": 0.2 * ls,
         }
         return mesh_sizes
+
+
+class NonMatchingSquareDomainOrthogonalFractures(SquareDomainOrthogonalFractures):
+    """Create a non-matching mixed-dimensional grid of a square domain with up to two
+    orthogonal fractures.
+
+    The setup is similar to :class:`SquareDomainOrthogonalFractures`, but the
+    geometry allows for non-matching grids and different resolution for each grid.
+    """
+
+    def set_geometry(self) -> None:
+        """Define geometry and create a non-matching mixed-dimensional grid.
+
+        We here make a non-matching mixed-dimensional grid in the sense that neither of
+        the fracture grids, interface grids or the rock grid are matching. This is done
+        by refining the fracture and interface grids.
+
+        """
+
+        # The fracture grids are replaced by a refined version of the fractures which
+        # are already present in the grid. The interface grids are replaced by first
+        # creating a new mixed-dimensional grid with a higher resolution. The new
+        # mixed-dimensional grid will "donate" its interface grids to the original
+        # mixed-dimensional grid.
+
+        # First set the geometry and create a matching mixed-dimensional grid.
+        assert isinstance(self, pp.ModelGeometry)
+        super().set_geometry()  # type:ignore[safe-super]
```

Comment:
@IvarStefansson please consider the following in particular: The call to set_geometry will generate a grid, which we then disregard in the below call. This can become costly if the grid is fine, while for standard testing, the cost will be negligible. I see three solutions:
1. Ignore the extra cost.
2. Replace the call to super().set_geometry() with the code contained within that method (call self.set_domain(), etc.). This is in effect a reimplementation of set_geometry(), and is risky if we ever change the super method.
3. Introduce a new method in model.geometry, create_grid(), and change the present method to reimplement only that one.

What do you think?

Reply:
I'm inclined towards 3.

Reply:
Agree. I will implement.